### ISSUE TITLE
"fetch" is not defined

### ISSUE THREAD

**Issue by lucasfeliciano** [eslint/eslint#4015]
`import 'isomorphic-fetch';`

expose a function called "fetch" as so react-native projects.

which rule i need to use to suppress this error just for this case?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
As this is react-specific feature, you might want to create an issue in [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react) repository. Or they might already support this (not sure). ESLint doesn't support react out of the box.

**Comment by lucasfeliciano** [comment]
I tried to use the plugin, but this is a feature from react-native so it is not implemented.

Besides that other modules could export that function, like  `import 'isomorphic-fetch';`  does

**Comment by btmills** [comment]
If you just want to make sure ESLint knows about the global variable, check out the [Specifying Globals](http://eslint.org/docs/user-guide/configuring#specifying-globals) section of our configuring docs. Will `/*global fetch:false*/` work?

**Comment by lucasfeliciano** [comment]
But I need to add this comment in every file that I use the 'fetch()', right?

I was looking for a rule to put in `.eslintrc`

**Comment by btmills** [comment]
Skip down a line in the docs, how about `"globals": { "fetch": false }`?

**Comment by ilyavolodin** [comment]
@lucasfeliciano Does @btmills suggestion helped?

**Comment by lucasfeliciano** [comment]
Hey guys, I'm traveling in brazil right now and didn't have a chance to
test.
I'll do this when I have a chance to close the ticket.
:)
Thanks for the help until now

**Comment by nzakas** [comment]
OK, closing this issue as there doesn't appear to be any task here.

**Comment by SimenB** [comment]
`fetch` is a global though, isn't it? https://developer.mozilla.org/en/docs/Web/API/Fetch_API

`isomorphic-fetch` is just a polyfill, IIRC. Whether `fetch` should be a predefined global I don't know, but it's not a react specific feature at all, unless I've misunderstood something

**Comment by sarkistlt** [comment]
same issue, in comment it works in config not, what the problem?

**Comment by juliusosokinas** [comment]
Set the environment for a browser inside the `.eslintrc` file:

```
env:
  browser: true
```

**Comment by divyanshu013** [comment]
Just adding to @julius-mfc, the following will work in a non-yaml fashion:

```
"env": {
    "browser": true
  }
```

**Comment by ajostergaard** [comment]
Any response to @SimenB 's comment?

**Comment by SimenB** [comment]
Wow, 2015.

The Ghost and @divyanshu013 are correct though, use `browser` env, then `fetch` is defined 😄

**Comment by chee** [comment]
it might be preferable to use 

```js
/* global fetch:false */
```

at the top of your file

or

```json
"globals": {
  "fetch": false
}```

in your config, so as not to pull in other globals from the `browser` env.

**Comment by ajostergaard** [comment]
Thanks @SimenB but the question you asked wasn't how to get around the issue, the question was why we have to get around it at all given that fetch is a global - no?!?

**Comment by SimenB** [comment]
`fetch` is only a global in browsers, not in Node. Globals without specifying anything else is (mostly) just language globals, not api globals. It's the same reason you have to specify `node` in env to use `require`, but not `import`

**Comment by ajostergaard** [comment]
Ahh - thanks! :) So really the react plugin should handle this - shouldn't it?

**Comment by SimenB** [comment]
No, not unless React automatically polyfills `fetch` in every environment (which it doesn't). Either tell eslint your code runs in a browser, or explicitly tell it fetch is a global.

**Comment by ajostergaard** [comment]
So really React Native, which does provide `fetch`, should also provide a plugin or at least document the 'integration'. :)

**Comment by platinumazure** [comment]
I think this would be an excellent discussion to have in the eslint-plugin-react and/or react-native repositories. :wink:

**Comment by maulikdhameliya** [comment]
"env": {
  "browser": true
},
  "globals": {
      "fetch": true
    }

**Comment by chee** [comment]
@maulikdhameliya you'll probably want `"fetch": false`
the boolean is about whether or not the global variable should be assignable in your code

as in

```js
// without fetch in globals
fetch = 'lol' // fetch is not defined

// with fetch: true
fetch = 'lol' // fine

// with fetch: false
fetch = 'lol' // error (if using `no-global-assign` (enabled in eslint:recommended))
```

**Comment by leoskyrocker** [comment]
From what I'm reading, react-native developers should prefer using
```"globals": {"fetch": false}```
over
```"env": {"browser": true}```
because apparently react-native is not 100% equivalent to a browser env.
One example is the atob function which is not in react-native's global, but is
in the list of [browser globals](https://github.com/sindresorhus/globals/blob/97c71054d75144c7ec58c6a0c2496ff90838fea3/globals.json#L237).
That means by specifying the browser env, we will also pull in variables that are not available in RN.

**Comment by SimenB** [comment]
For RN you could also go for something like https://github.com/satya164/eslint-plugin-react-native-globals or potentially open up at issue at https://github.com/sindresorhus/globals so you could do `{"env": "react-native": true}`.

**Comment by j-f1** [comment]
> or potentially open up at issue at https://github.com/sindresorhus/globals so you could do `{"env": "react-native": true}`.

https://github.com/sindresorhus/globals/issues/82
